evalml/pipelines/components/transformers/preprocessing/time_series.py

```diff
@@ -0,0 +1,42 @@
+import pandas as pd
+
+from evalml.pipelines.components.transformers.transformer import Transformer
+
+
+class DelayedFeaturesTransformer(Transformer):
+    name = "Delayed Features Transformer"
+    hyperparameter_ranges = {}
+    needs_fitting = False
+
+    def __init__(self, max_delay=2, random_state=0, **kwargs):
```

Comment:
In the future, we may want to add a parameter for features that should not be lagged (they don't change with time). I think proceeding without that functionality is good enough for an MVP though.